pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -737,6 +737,11 @@ class Timestamp(_Timestamp):
         """
         from pandas import Period
 
+        if self.tz is not None:
+            # GH#21333
+            warnings.warn("Converting to Period representation will "
+                          "drop timezone information.")
```

Comment:
Warning type and stack level?

Reply:
This questions applies to everywhere where you placed warnings.

Reply:
No idea what warning type to use.  Suggestions?

As to stack level, I tried a bunch to get that to work with tm.assert_produces_warning and eventually threw in the towel.

Reply:
Since the warnings are raised in direct user-api methods, normally putting a `stacklevel=2` should do the correct thing. 
What did you not get working in the tests? (which kind of code sample)

Reply:
> What did you not get working in the tests?

IIRC the problem is that the affected code paths issues both the new `FutureWarning` and in some cases also a `PerformanceWarning`.  `tm.assert_produces_warning` doesn't support multiple expected warnings, and my attempt to modify it led to tests failing the stacklevel checks.